src/components/Tool/FixMailingAddresses/FixMailingAddresses.tsx

```diff
@@ -150,17 +150,53 @@ const FixMailingAddresses: React.FC<Props> = ({ accountListId }: Props) => {
       });
     });
     setSourceOptions([...existingSources]);
+
+    const newDataState = data
+      ? data.contacts.nodes?.reduce(
+          (map, contact) => ({
+            ...map,
+            [contact.id]: {
+              addresses: contact.addresses.nodes.map((address) => {
+                existingSources.add(address.source);
+                return { ...address };
+              }),
+            },
+          }),
+          {},
+        )
+      : {};
+
+    setDataState(newDataState);
   }, [loading, data]);
 
+  const handleChangePrimary = (
+    contactId: string,
+    numberIndex: number,
```

Comment:
It might be worth since we have access to the `address.id` to use that instead. What way if there was ever a case where the index was changed, there wouldn't be a bug.